python/reg_interface/scripts/sca.py

```diff
@@ -8,37 +8,105 @@
 from reg_utils.reg_interface.common.sca_utils import *
 from reg_utils.reg_interface.common.sca_common_utils import *
 from time import *
-import array
-import struct
 import socket
 
-def main():
-
-    instructions = ""
-    ohMask = 0
-    ohList = []
-
-    if len(sys.argv) < 4:
-        print('Usage: sca.py <card_name> <oh_mask> <instructions>. If running on the card, put `local` instead of hostname')
-        print('instructions:')
-        print('  r:        SCA reset will be done')
-        print('  h:        FPGA hard reset will be done')
-        print('  hh:       FPGA hard reset will be asserted and held')
-        print('  fpga-id:  FPGA ID will be read through JTAG')
-        print('  sysmon:   Read FPGA sysmon data repeatedly')
-        print('  program-fpga:   Program OH FPGA with a bitfile or an MCS file. Requires a parameter "bit" or "mcs" and a filename')
+def compareFwFiles(args):
+    if ((args.fwFileBit is None) or (args.fwFileMCS is None)):
+        print("Usage: sca.py <cardName> <ohMask> compare-mcs-bit --fwFileMCS=<mcs_filename> --fwFileBit=<bit_filename>")
```

Comment:
Probably last set of comments on this...
Given that previously these were positional, I'd probably keep them that way (i.e., treat them as args rather than required options)
Then the syntax handling is done by `argparse`, and the help message can be put into the parser, rather than in the function

This last comment would basically hold for all other functions too

Reply:
I think this is a bit excessive.  If this is desired it should be implemented in a different PR, and by either the original author @evka85 or @mexanick who refactored this from `xhal` to begin with.  The goal of this PR was to resolve a critical bug that prevented testing and QC of the OHv3c's and I have done that; with extensive testing to illustrate that it works.

Reply:
Additionally since you've gone back and forth a few times asking for sometimes contradictory requests.  If I understand this comment you are basically asking me to undo the previous request of dropping `--fwFile` for the `fpga-program` command in favor of a positional argument.

If this is the case in the future I think it would be better to just put forth all requests upfront, and clearly explained, rather than back and forth action which wastes both our time (mine in programming, and yours in reviewing).

Reply:
Well, it is **this** PR that is proposing a change in the current behaviour... ergo, the onus is on this one to be done correctly.

>  If I understand this comment you are basically asking me to undo the previous request of dropping --fwFile for the fpga-program command in favor of a positional argument.

You wouldn't drop `fwFile`, it would change from its current modified implementation of a "flagged required *option*", back to its original implementation as an *argument*, which is what my initial request was for, (it additionally reduces the typing of the user, which has often been your motivation for previous changes)

I wouldn't say I've gone back and forth... I'd simply say that I worked from the baseline of the initial change, saw that there was significant room for improvement, and added further refinements. I would have preferred that everything was modified in this way to begin with.

Reply:
> I would have preferred that everything was modified in this way to begin with.

Then to alleviate frustration in the future, and prevent lost time, you should be more descriptive in your feedback on what you would like to see, or be less restrictive on what is pushed into central.  From your comments [here](https://github.com/cms-gem-daq-project/reg_utils/pull/22#issuecomment-404862393) there was no logical way for someone to  infer that someone should come up with `sub_commands` as later mentioned [here](https://github.com/cms-gem-daq-project/reg_utils/pull/22#discussion_r202934788).  And from your example [here](https://github.com/cms-gem-daq-project/reg_utils/pull/22#discussion_r202926297) shows specifically the type of functionality I have provided.

If you wanted to maintain positional arguments and work with subcommands from the very beginning, you should have explained this; and I would have done it, from the beginning.  But this back and forth over 5 days causes a loss of time which I have asked you to try to avoid in the past.